tls-client/main.cpp

```diff
@@ -256,22 +261,31 @@ class HelloHTTPS {
 
 
         /* Read data out of the socket */
-        ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer, sizeof(_buffer));
+        int read = 0;
+        do {
+            ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer + read, 
+                                   sizeof(_buffer) - 1);
+            if (ret > 0)
+              read += ret;
+            else if (ret != MBEDTLS_ERR_SSL_WANT_READ &&
+                ret != MBEDTLS_ERR_SSL_WANT_WRITE)
+              break;
+            /* Check each of the flags */
+            _got200 = _got200 || strstr(_buffer, HTTPS_OK_STR) != NULL;
+            _gothello = _gothello || strstr(_buffer, HTTPS_HELLO_STR) != NULL;
+        } while (!_got200 || !_gothello);
         if (ret < 0) {
-            if (ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
-                print_mbedtls_error("mbedtls_ssl_read", ret);
-                onError(_tcpsocket, -1 );
-            }
+            print_mbedtls_error("mbedtls_ssl_read", ret);
             delete[] buf;
+            _tcpsocket->close();
             return;
         }
-        _bpos = static_cast<size_t>(ret);
+        _bpos = static_cast<size_t>(read);
 
         _buffer[_bpos] = 0;
-
-        /* Check each of the flags */
-        _got200 = _got200 || strstr(_buffer, HTTPS_OK_STR) != NULL;
-        _gothello = _gothello || strstr(_buffer, HTTPS_HELLO_STR) != NULL;
+        
```

Comment:
There are whitespace characters on this empty line, could you please remove them?